scaleway/resource_compute_instance_volume.go

```diff
@@ -0,0 +1,203 @@
+package scaleway
+
+import (
+	"fmt"
+
+	"github.com/dustin/go-humanize"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	namesgenerator "github.com/scaleway/scaleway-sdk-go/namegenerator"
+)
+
+func resourceScalewayComputeInstanceVolume() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayComputeInstanceVolumeCreate,
+		Read:   resourceScalewayComputeInstanceVolumeRead,
+		Update: resourceScalewayComputeInstanceVolumeUpdate,
+		Delete: resourceScalewayComputeInstanceVolumeDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "the name of the volume",
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if new == "" {
+						return true
+					}
+					return old == new
+				},
+			},
+			"size": {
```

Comment:
Quick suggestion: use `size_in_gb`. Ultimately the size is always in GB (I used the same naming for root_rolume on instance)